int16[] min
int16[] max
---
bool success
string message